cpp/src/arrow/util/future.h

```diff
@@ -48,7 +49,19 @@ template <typename T>
 struct is_future<Future<T>> : std::true_type {};
 
 template <typename Signature>
-using result_of_t = typename std::result_of<Signature>::type;
+using result_of_t_2 = typename std::result_of<Signature>::type;
```

Comment:
Is this actually used?

Reply:
No, sorry- I was using this during debugging. I'll remove it